### DISCUSSION TITLE
Is it ill-advised to extend `KafkaListenerAnnotationBeanPostProcessor` in one's own application?

### DISCUSSION THREAD

**Discussion by igorflyunt** [spring-projects/spring-kafka#3018] [Q&A] [answered]
I want to set a custom group id and listener id names based on a combination of beanName and topic.
Would you say having KafkaListenerAnnotationBeanPostProcessor extended in my own application is a right thing to do?
If it's ok to extend BPP, how to do this properly? 
What are the alternatives?

**Comment by artembilan** [comment, marked as answer]
Take a look into `ContainerCustomizer` instead which you can inject into the `AbstractKafkaListenerContainerFactory`.

**Reply by igorflyunt** [reply]
Oh, calling `setBeanName` will actually set a listener id; And using ...`getContainerPropeties` to retrieve a `messageListener` obj gives me a bean name of the actual class with the `KafkaListener` annotated methods.
Not seeing this the first time was the only reason I'd decided against the usage of `ContainerCustomizer`.

Anyway, thanks for taking a time to give such a prompt reply)
